src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/AssessmentController.kt

```diff
@@ -66,35 +64,25 @@ class AssessmentController(
     val user = userService.getUserForRequest()
     val resolvedSortDirection = sortDirection ?: SortDirection.asc
     val resolvedSortBy = sortBy ?: AssessmentSortField.assessmentArrivalDate
+    val domainSummaryStatuses = statuses?.map { assessmentTransformer.transformApiStatusToDomainSummaryState(it) } ?: emptyList()
 
     val (summaries, metadata) = when (xServiceName) {
       ServiceName.cas2 -> throw UnsupportedOperationException("CAS2 not supported")
       ServiceName.temporaryAccommodation -> {
-        val summaries = if (crn != null) {
-          assessmentService.getAssessmentSummariesByCrnForUserCAS3(user, crn, xServiceName)
-        } else {
-          assessmentService.getVisibleAssessmentSummariesForUserCAS3(user)
+        val (summaries, metadata) = assessmentService.getAssessmentSummariesForUserCAS3(
+          user,
+          crn,
+          xServiceName,
+          domainSummaryStatuses,
+          PageCriteria(resolvedSortBy, resolvedSortDirection, page, perPage ?: defaultPageSize),
```

Comment:
i think you can pass through a null perPage value and the default will be resolved in the PaginationUtils, as we did for CAS1 below (i.e. no need to set default here, and i think we can remove injection of pagination.default-page-size)

Reply:
done